tests/update/patch_before_migration.pm

```diff
@@ -73,13 +76,46 @@ sub patching_sle {
     set_var("SYSTEM_PATCHED", 1);
 }
 
+sub install_patterns {
+    select_console 'root-console';
+    my $pcm = 0;
+    my @pt_list;
+    # install all patterns from product.
+    if (check_var('PATTERNS', 'all')) {
+        @pt_list = split(/ /, script_output("zypper pt -u | grep '^ ' | tr -s ' ' | cut -d ' ' -f 3 | tr -s '\n' ' '"));
+    }
+    #install certain pattern from parameter.
+    else {
+        @pt_list = split(/,/, get_var('PATTERNS'));
+    }
+
+    for my $pt (@pt_list) {
+        # Cloud patterns are conflict by each other, only install cloud pattern from single vender.
+        if ($pt =~ /Amazon-Web-Services|Google-Cloud-Platform|Microsoft-Azure/) {
+            if ($pcm == 0) {
+                zypper_call "in -t pattern Amazon-Web-Services*"   if ($pt =~ /Amazon-Web-Services/);
+                zypper_call "in -t pattern Microsoft-Azure*"       if ($pt =~ /Microsoft-Azure/);
+                zypper_call "in -t pattern Google-Cloud-Platform*" if ($pt =~ /Google-Cloud-Platform/);
+                $pcm = 1;
```

Comment:
There would be conflict if any cloud pattern has been installed on the hdd image.

Reply:
This part looks redundant. Could we simply pass the petterns that need to be installed based on variable `SCC_ADDON=pcm,asm,...`?